pkg/config/config.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"flag"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var config poseidonConfig
+
+type poseidonConfig struct {
+	SchedulerName      string `json:"schedulername,omitempty"`
+	FirmamentAddress   string `json:"firmamentaddress,omitempty"`
+	KubeConfig         string `json:"kubeConfig,omitempty"`
+	KubeVersion        string `json:"kubeVersion,omitempty"`
+	StatsServerAddress string `json:"statsServerAddress,omitempty"`
+	SchedulingInterval int    `json:"schedulingInterval,omitempty"`
+	FirmamentPort      string `json:"firmamentPort,omitempty"`
+	ConfigPath         string `json:"configPath,omitempty"`
+}
+
+// GetSchedulerName returns the SchedulerName from config
+func GetSchedulerName() string {
+	return config.SchedulerName
+}
+
+// GetFirmamentAddress returns the FirmamentAddress from config
+func GetFirmamentAddress() string {
+	// join the firmament address and port with a colon separator
+	// Passing the firmament address with port and colon separator throws an error
+	// for conversion from yaml to json
+	values := []string{config.FirmamentAddress, config.FirmamentPort}
+	return strings.Join(values, ":")
+}
+
+// GetKubeConfig returns the KubeConfig from config
+func GetKubeConfig() string {
```

Comment:
As this function simply return the `KubeConfig` field. I don't see any benefit to create a function. When the `poseidonConfig` becomes bigger, the 1:1 functions will grows as well.

Reply:
Yes having a getter for all the fields is not necessary, but I added for completeness because of the other methods like the ```'GetFirmamentAddress'``` and the ```'GetKubeVersion'``` have some logic in it.
We can remove for other methods ```'GetStatsServerAddress'```, ```'GetSchedulingInterval'```  ```'GetConfigPath'``` ```'GetSchedulerName'```.

But lets do it for new fields added in the future.
Also we should completely remove the commad line flags and use config files from next release.
We can discuss on this.